### ISSUE TITLE
3.2.4 to 3.2.5 org.apache.tomcat.jdbc.pool.PoolExhaustedException

### ISSUE THREAD

**Issue by AlexKovynev** [grails/grails-core#10445]
Unable to fetch a connection in 30 seconds Timeout: Pool empty and other blabla.

I am upgrade also gorm to 6.0.6 from 6.0.5 and hibernate from 5.1.3 to 5.1.4.
It happens after some hours. I don't know how reproduce in test app:(

**Comment by graemerocher** [comment]
Tried to reproduce in a simple app but cannot either. We will need more information. Are you using `withNewSession`? Are you using `@Transactional` around all GORM operations or do you have some without?

A sample application would be good.

**Comment by AlexKovynev** [comment]
I use withSession in quartz job. Not @Transactional or withNewSession. Application is old, A test for what? If i knew... Return to 3.2.3. All fine, maybe some one else with very high load post some details. For me small shop fell in above four hours. Gorm 6.0.6 and hibernate 5.1.4, so maybe with this https://github.com/grails/grails-core/issues/10444 is same roots.

**Comment by graemerocher** [comment]
@AlexKovynev Can you please try declare the following in `build.gradle`:

```groovy
     compile "org.grails:grails-datastore-gorm-hibernate5:6.0.7.BUILD-SNAPSHOT"
```

And let me know if you are still seeing the issue

**Comment by AlexKovynev** [comment]
For me all work with 6.0.6, but with 3.2.3, ok, i'l try

**Comment by AlexKovynev** [comment]
Unfortunately all bad. I annotated all my controller, use snapshot 6.0.7 (i check war content), all down very quickly, above minute.

**Comment by AlexKovynev** [comment]
How can i set tomcat pool size? Can you give recommendations on it value? Now i understood, actions begin to use many transactions instead one or something.  On the other hand, I do not understand, the pool is full, after the fall, it shall be clean, connection should be closed, and this does not happen.

**Comment by graemerocher** [comment]
I have been able to reproduce the issue and I am looking into it. Thanks for the report

**Comment by graemerocher** [comment]
@AlexKovynev I have committed a fix, please try the latest `6.0.7.BUILD-SNAPSHOT` snapshot

**Comment by AlexKovynev** [comment]
Still working. But i have some slowness in quartz jobs. Look like each query use each own transaction and commit them as i set AUTO to hibernate flush.
Wait for generate-* commands recovery:) How can I begin one transaction for many tables? Or i can use any of used tables for this?
